pkg/storage/chunk/storage/caching_index_client_test.go

```diff
@@ -92,6 +95,7 @@ func TestTempCachingStorageClient(t *testing.T) {
 		return true
 	})
 	require.NoError(t, err)
+	time.Sleep(1000 * time.Millisecond)
```

Comment:
Same suggestion as above to do `assert.Eventually`. I think it would apply to most of the cases below.
We just want to make sure the write back queue is cleared since it is async now.

Reply:
done，thanks